src/hypothesis/strategies.py

```diff
@@ -982,32 +987,61 @@ def fractions(min_value=None, max_value=None, max_denominator=None):
 
     If max_denominator is not None then the absolute value of the denominator
     of any generated values is no greater than max_denominator. Note that
-    max_denominator must be at least 1.
+    max_denominator must be None or a positive integer, and the denominator
```

Comment:
Yeah, agreed. I hadn't even realised/remembered that we supported non-integer bounds to integers, but we should definitely handle the case where this results in unsatisfiable bounds correctly. Anything where an AssertionError escapes into the wild is a bug.